zaf/zaf/component/factory.py

```diff
@@ -144,6 +152,10 @@ def call(self, callable, scope, *args, fixated_entities=None, extra_req=None, **
         fixated_entity_mapping = self._component_manager.component_name_to_entity_mapping(
             fixated_entities)
 
+        for name in pre_instantiated or {}:
+            instance = pre_instantiated[name]
+            scope.register_instance(_InstanceId(instance.__class__, args, kwargs), instance)
```

Comment:
I'm not 100% sure about this line, but it works :shrug:

Reply:
What problem will this solve?

Reply:
Hmm... For some reason I can't reproduce the issue now :confused: I kept hitting this RunnerComponentException in _k2/k2/runner/components.py_:
```python
    def __init__(self, test_case_definition=None):
        if test_case_definition is None:
            raise RunnerComponentException(
                'The TestContext component can only be pre-instantiated by the test runner and can only be used '
                'on the test scope.')
```

Reply:
This is how the call to the component factory looks like:
```python
        def run_test_case(*args, **kwargs):
...
                            self.component_factory.call(
                                current_test_case.run,
                                scope,
                                *args,
                                extra_req=extra_req,
                                pre_instantiated=self.create_pre_instantiated_components(
                                    current_test_case),
                                **kwargs)
...
    def create_pre_instantiated_components(self, test_case_definition):
        """Create the pre-instantiated TestContext instance."""
        return {'TestContext': TestContext(test_case_definition)}
```

Reply:
Commit removed in latest diff. Issue is fixed by upstream changes =)